File: 서인/CH06/6_공주_구하기.js
function solution(n, k){
    let answer;
    let q = [...Array(n)].map((v,i) => i+1);

    let cnt = 1;
    while (q.length >= 2){
      if (cnt !== k){
        cnt++;
        let tmp = q[0]
        q.splice(0,1)
        q.push(tmp)
      } else {
        cnt = 1;
        q.splice(0,1)
      }
    }
    answer = q[0]
    return answer;
}

console.log(solution(8, 3));

File: 종복/Java/Stack&Queue/6. 공주 구하기.java
import java.util.*;
class Main {	
	
    public int solution(int n, int k){
		int answer=0;
		
		Queue<Integer> q = new LinkedList<Integer>();
		
		for(int i=1; i<=n; i++){
		    q.add(i);
		}
		
		while(true){
		    
		    if(q.size() == 1){
		        break;
		    }
		    
		    for(int i=1; i<=k; i++){
		        int num = q.poll();
		        if(i != k){
		            q.add(num);
		        }
		    }
		}
		
	    answer = q.poll();  	
      
		return answer;
	}
	public static void main(String[] args){
		Main T = new Main();
		Scanner kb = new Scanner(System.in);
		int n=kb.nextInt();
		int k=kb.nextInt();
		System.out.println(T.solution(n, k));
	}
}
